boss-ws/src/main/java/org/genomebridge/boss/http/BossApplication.java

```diff
@@ -95,11 +99,12 @@ public static BossAPI getAPI() {
     }
 
     private static ObjectStore getObjectStore( ObjectStoreConfiguration config ) throws Exception {
-        if ( "S3".equals(config.type) )
-            return new S3ObjectStore(config);
-        if ( "GCS".equals(config.type) )
-            return new GCSObjectStore(config);
-        throw new IllegalStateException("ObjectStore configuration has unrecognized type: "+config.type);
+        switch ( config.type ) {
+        case S3: return new S3ObjectStore(config);
+        case GCS: return new GCSObjectStore(config);
+        case FCS: return new FCSObjectStore(config);
+        }
+        throw new IllegalArgumentException("No handler for ObjectStoreType "+config.type.toString());
```

Comment:
Can this ever be reached, since it's not a vaild ObjectStoreType enum value?  My guess is that the failure would occur at the time of enum value creation instead.